cpp/src/neighbors/tiered_index.cu

```diff
@@ -175,6 +175,30 @@ void search(raft::resources const& res,
     res, search_params, ivf_pq::typed_search, queries, neighbors, distances, sample_filter);
 }
 
+auto merge(raft::resources const& res,
+           const std::vector<tiered_index::index<cagra::index<float, uint32_t>>*>& indices)
+  -> tiered_index::index<cagra::index<float, uint32_t>>
```

Comment:
I think ideally we would have merge() return index<type of the input> and, as you have done, use the PIMPL idom to allow the return of either the same index type that was input, or a composite_index.